### PR TITLE
chore: manually assign custom domain referrer [OTE-516]

### PR DESCRIPTION
**Pull request by yogurtandjam**
context: https://dydx-team.slack.com/archives/C03JV4VJQ6N/p1716565422932799

manually setting a domain referrer since the one automatically it often returns none.
This will help us know if a domain referrer is simply not being provided or if a race condition/setting is causing amplitude to fail to capture this.

also removes the `referrer` event property for 2 reasons:
1. it tracks utm_source, which amplitude already tracks
2. it has no business being an event property. even if we want to add this back at some point it should be as a user property

### CODE REVIEW

File: src/lib/analytics.ts [hunk 1]
```diff
@@ -26,13 +24,12 @@ export const identify = (property: AnalyticsUserProperty) => {
 };
 
 export const track = (event: AnalyticsEvent) => {
-  const eventDataWithReferrer = { ...(event.payload ?? {}), referrer: testFlags.referrer };
```
Comment (rosepuppy) [inline src/lib/analytics.ts:29]: why is this removed?

Comment (rosepuppy) [reply src/lib/analytics.ts:29]: oh because referrer is automatically tracked by amplitude?

**Review by rosepuppy — APPROVED** [review]

File: src/lib/analytics.ts (hunk 1 above)
Comment (yogurtandjam) [reply src/lib/analytics.ts:29]: yup! utm_source is already tracked by amplitude and i have a feeling it's actually slightly more accurate, so i'm just getting rid of it to reduce confusion
